modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientHandlerMetricSource.java

```diff
@@ -0,0 +1,399 @@
+    }
+
+    void requestsActiveDecrement() {
+        Holder h = holder();
+
+        if (h != null) {
+            h.requestsActive.decrement();
+        }
+    }
+
+    /**
+     * Gets processed requests.
+     *
+     * @return Processed requests.
+     */
+    public long requestsProcessed() {
+        Holder h = holder();
+
+        return h == null ? 0 : h.requestsProcessed.value();
+    }
+
+    void requestsProcessedIncrement() {
+        Holder h = holder();
+
+        if (h != null) {
+            h.requestsProcessed.increment();
+        }
+    }
+
+    /**
+     * Gets failed requests.
+     *
+     * @return Failed requests.
+     */
+    public long requestsFailed() {
+        Holder h = holder();
+
+        return h == null ? 0 : h.requestsFailed.value();
+    }
+
+    void requestsFailedIncrement() {
+        Holder h = holder();
+
+        if (h != null) {
+            h.requestsFailed.increment();
+        }
+    }
+
+    /**
+     * Gets active transactions.
+     *
+     * @return Active transactions.
+     */
+    public long transactionsActive() {
+        Holder h = holder();
+
+        return h == null ? 0 : h.transactionsActive.value();
+    }
+
+    /**
+     * Increments active transactions.
+     */
+    public void transactionsActiveIncrement() {
+        Holder h = holder();
+
+        if (h != null) {
+            h.transactionsActive.increment();
+        }
+    }
+
+    /**
+     * Decrements active transactions.
+     */
+    public void transactionsActiveDecrement() {
+        Holder h = holder();
+
+        if (h != null) {
+            h.transactionsActive.decrement();
+        }
+    }
+
+    /**
+     * Gets active cursors.
+     *
+     * @return Active cursors.
+     */
+    public long cursorsActive() {
+        Holder h = holder();
+
+        return h == null ? 0 : h.cursorsActive.value();
+    }
+
+    /**
+     * Increments active cursors.
+     */
+    public void cursorsActiveIncrement() {
+        Holder h = holder();
+
+        if (h != null) {
+            h.cursorsActive.increment();
+        }
+    }
+
+    /**
+     * Decrements active cursors.
+     */
+    public void cursorsActiveDecrement() {
+        Holder h = holder();
+
+        if (h != null) {
+            h.cursorsActive.decrement();
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    protected void init(MetricSetBuilder bldr, Holder holder) {
+        holder.register(bldr);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    protected Holder createHolder() {
+        return new Holder();
+    }
+
+    /**
+     * Holder.
+     */
+    protected static class Holder implements AbstractMetricSource.Holder<Holder> {
+        private final AtomicLongMetric connectionsInitiated =
+                new AtomicLongMetric("connections.Initiated", "Total initiated connections");
+
+        private final AtomicLongMetric sessionsAccepted =
+                new AtomicLongMetric("sessions.Accepted", "Total accepted sessions");
```

Comment:
IMHO: redundant grouping here. The amount of metrics is not large so there is no need to additional grouping like `sessions.Active` and `sessions.Rejected`. 

So my proposal is to rename metrics to something l ike `SessionsActive`, `SessionsRejected`, `BytesSent`, etc.

Reply:
Done.